crates/egui_demo_app/src/apps/image_viewer.rs

```diff
@@ -0,0 +1,212 @@
+        egui::TopBottomPanel::new(TopBottomSide::Top, "url bar").show(ctx, |ui| {
+            ui.horizontal_centered(|ui| {
+                ui.label("URI:");
+                ui.text_edit_singleline(&mut self.uri_edit_text);
+                if ui.small_button("✔").clicked() {
+                    ctx.forget_image(&self.current_uri);
+                    self.uri_edit_text = self.uri_edit_text.trim().to_owned();
+                    self.current_uri = self.uri_edit_text.clone();
+                };
+
+                #[cfg(not(target_arch = "wasm32"))]
+                if ui.button("file…").clicked() {
+                    if let Some(path) = rfd::FileDialog::new().pick_file() {
+                        self.uri_edit_text = format!("file://{}", path.display());
+                        self.current_uri = self.uri_edit_text.clone();
+                    }
+                }
+            });
+        });
+
+        egui::SidePanel::new(Side::Left, "controls").show(ctx, |ui| {
+            // uv
+            ui.label("UV");
+            ui.add(Slider::new(&mut self.image_options.uv.min.x, 0.0..=1.0).text("min x"));
+            ui.add(Slider::new(&mut self.image_options.uv.min.y, 0.0..=1.0).text("min y"));
+            ui.add(Slider::new(&mut self.image_options.uv.max.x, 0.0..=1.0).text("max x"));
+            ui.add(Slider::new(&mut self.image_options.uv.max.y, 0.0..=1.0).text("max y"));
+
+            // rotation
+            ui.add_space(2.0);
+            let had_rotation = self.image_options.rotation.is_some();
+            let mut has_rotation = had_rotation;
+            ui.checkbox(&mut has_rotation, "Rotation");
+            match (had_rotation, has_rotation) {
+                (true, false) => self.image_options.rotation = None,
+                (false, true) => {
+                    self.image_options.rotation =
+                        Some((Rot2::from_angle(0.0), Vec2::new(0.5, 0.5)));
+                }
+                (true, true) | (false, false) => {}
+            }
+
+            if let Some((rot, origin)) = self.image_options.rotation.as_mut() {
+                let mut angle = rot.angle();
+
+                ui.label("angle");
+                ui.drag_angle(&mut angle);
+                *rot = Rot2::from_angle(angle);
+
+                ui.add(Slider::new(&mut origin.x, 0.0..=1.0).text("origin x"));
+                ui.add(Slider::new(&mut origin.y, 0.0..=1.0).text("origin y"));
+            }
+
+            // bg_fill
+            ui.add_space(2.0);
+            ui.label("Background color");
+            ui.color_edit_button_srgba(&mut self.image_options.bg_fill);
+
+            // tint
+            ui.add_space(2.0);
+            ui.label("Tint");
+            ui.color_edit_button_srgba(&mut self.image_options.tint);
+
+            // aspect ratio
+            ui.add_space(2.0);
+            ui.checkbox(&mut self.maintain_aspect_ratio, "Maintain aspect ratio");
+
+            // fit
```

Comment:
The ui for the fitting is not very intuitive. It would be good to explain how the fitting code works (is it the minimum of fraction and max size?)